Update CONTRIBUTING.md with tips for successful PRs (#2598)
WordPress/WordPress-Coding-Standards: .github/CONTRIBUTING.md

* **Respect copyright and licensing.**
  Only submit code that you have written yourself or that comes from sources where the license clearly allows inclusion. Submitting code that infringes on copyright or licensing terms puts both you and the project at legal risk, and such contributions cannot be accepted.

* **Do not submit AI-generated code.**
  Pull requests containing AI-generated code are not acceptable. Beyond copyright and licensing uncertainties, AI-generated contributions consistently require disproportionate amounts of maintainer time to review, correct, or rewrite. This wastes limited project resources and slows progress for everyone. Submitting AI-generated code may be treated as a violation of our [Code of Conduct](../CODE_OF_CONDUCT.md).

* **Focus on quality and clarity.**
  Take time to explain *why* the change is needed, and include tests or examples where appropriate. Clear, self-written explanations make it more straightforward for reviewers to understand what you are trying to achieve.

* **Think about long-term maintainability.**

>   Code should align with WordPress coding standards and be written in a way that others can readily read, understand, and maintain.

Reviewer: ```suggestion
  Code should align with WordPress Coding Standards and be written in a way that others can readily read, understand, and maintain.
```

* **Be collaborative.**

>   If you’re unsure about an approach, open an issue or draft PR first to start a conversation.

Reviewer: ```suggestion
  If you're unsure about an approach, open an issue first to start a conversation.
```

Curly quote replacement.

I agree with Juliette, I think it's better to open an issue (or search if the issue has been opened about the discussion topic), and then discuss it there. Opening a draft PR can just cause noise.

> By following these tips, you’ll save time for both yourself and the maintainers — and increase the likelihood that your contribution can be merged smoothly.

Reviewer: ```suggestion
By following these tips, you'll save time for both yourself and the maintainers — and increase the likelihood that your contribution can be merged smoothly.
```

Another curly quote replacement.
